packages/containers/src/HeaderBar/HeaderBar.sagas.js

```diff
@@ -0,0 +1,47 @@
+import cmf from '@talend/react-cmf';
+import { call, put, takeEvery, take } from 'redux-saga/effects';
+
+import Connected from './HeaderBar.connect';
+import Constants from './HeaderBar.constant';
+
+/**
+ * This saga takes care of fetching authorized products for the HeaderBar
+ * container according to the provided products URL in the action's payload.
+ * @param {Object} action
+ */
+export function* fetchProducts(action) {
+	const { url, lang } = action.payload;
+	const productsUrl = `${url}?lang=${lang}`;
+
+	yield put(Connected.setStateAction({ fetchingProducts: true }));
+
+	const { response, data } = yield call(cmf.sagas.http.get, productsUrl);
+
+	yield put(Connected.setStateAction({ fetchingProducts: false }));
+
+	if (!response.ok) {
+		return; // Loading products failed, do nothing
+	}
+
+	yield put(cmf.actions.collections.addOrReplace(Constants.COLLECTION_ID, data));
+}
+
+/**
+ * Handle the opening of a product from the HeaderBar
+ * @param {Object} action
+ */
+export function handleOpenProduct(action) {
```

Comment:
this should live in a component, also this should be handled by a simple html link with a target blank

Reply:
no we have some case where we need more (check if a browser extension exists, open the extension or the market place)
That code will goes here later

Reply:
i think this info should land on a code comment for future readings and understandings